pin/pin.go

```diff
@@ -519,7 +520,9 @@ func (p *pinner) PinWithMode(c *cid.Cid, mode PinMode) {
 	}
 }
 
-func hasChild(ds mdag.LinkService, root *cid.Cid, child *cid.Cid) (bool, error) {
+// hasChild recursively looks for a Cid among the children of a root Cid.
+// The visit function can be used to shortcut already-visited branches.
+func hasChild(ds mdag.LinkService, root *cid.Cid, child *cid.Cid, visit func(*cid.Cid) bool) (bool, error) {
```

Comment:
At least change the **hasChild** func format in someting like this.
```go
func hasChild(
ds mdag.LinkService,
root *cid.Cid,
child *cid.Cid,
visit func(*cid.Cid) bool,
) (bool, error) {
//body
}
```

This should be documented somewhere to enforce a number of columns a func can have.